drake/solvers/test/testOptimizationProblem.cpp

```diff
@@ -252,28 +257,10 @@ void multiLCP() {
 
   prog.addLinearComplementarityConstraint(M, q, {x});
   prog.addLinearComplementarityConstraint(M, q, {y});
-  prog.solve();
+  EXPECT_NO_THROW(prog.solve());
   prog.printSolution();
-  valuecheckMatrix(x.value(), Vector2d(16, 0), 1e-4);
-  valuecheckMatrix(y.value(), Vector2d(16, 0), 1e-4);
+  EXPECT_NO_THROW(valuecheckMatrix(x.value(), Vector2d(16, 0), 1e-4));
+  EXPECT_NO_THROW(valuecheckMatrix(y.value(), Vector2d(16, 0), 1e-4));
 }
 
-int main(int argc, char* argv[]) {
-  // SNOPT tests
-  try {
-    testAddFunction();
-    trivialLeastSquares();
-    sixHumpCamel();
-    gloptipolyConstrainedMinimization();
-  } catch (const std::exception& e) {
-    // If the exception is SNOPT unavailble, skip the remaining snopt tests
-    // and proceed; if not, reraise it to fail.
-    if (std::string(e.what()) != "SNOPT unavailable") {
-      throw;
-    }
-  }
-  simpleLCPConstraintEval();
-  simpleLCP();
-  multiLCP();
-  return 0;
 }
```

Comment:
No, my fault -  I thought this brace closed an anonymous namespace, but actually it closes main().